core/src/toga/widgets/mapview.py

```diff
@@ -0,0 +1,232 @@
+
+    @title.setter
+    def title(self, title: str) -> None:
+        self._title = str(title)
+        if self.interface:
+            self.interface._impl.update_pin(self)
+
+    @property
+    def subtitle(self) -> str | None:
+        "The subtitle of the pin."
+        return self._subtitle
+
+    @subtitle.setter
+    def subtitle(self, subtitle: str | None) -> None:
+        if subtitle is not None:
+            subtitle = str(subtitle)
+        self._subtitle = subtitle
+        if self.interface:
+            self.interface._impl.update_pin(self)
+
+
+class MapPinSet:
+    def __init__(self, interface, pins):
+        self.interface = interface
+        self._pins = set()
+
+        if pins:
+            for item in pins:
+                self.add(item)
+
+    def __repr__(self):
+        return f"<MapPinSet ({len(self)} pins)>"
+
+    def __iter__(self):
+        """Return an iterator over the pins on the map."""
+        return iter(self._pins)
+
+    def __len__(self):
+        """Return the number of pins being displayed."""
+        return len(self._pins)
+
+    def add(self, pin):
+        """Add a new pin to the map.
+
+        :param pin: The :any:`toga.MapPin` instance to add.
+        """
+        pin.interface = self.interface
+        self._pins.add(pin)
+        self.interface._impl.add_pin(pin)
+
+    def remove(self, pin):
+        """Remove a pin from the map.
+
+        :param pin: The  :any:`toga.MapPin` instance to remove.
+        """
+        self.interface._impl.remove_pin(pin)
+        self._pins.remove(pin)
+        pin.interface = None
+
+    def clear(self):
+        """Remove all pins from the map."""
+        for pin in self._pins:
+            self.interface._impl.remove_pin(pin)
+        self._pins = set()
+
+
+class OnSelectHandler(Protocol):
+    def __call__(self, widget: MapView, *, pin: MapPin, **kwargs: Any) -> None:
+        """A handler that will be invoked when the user selects a map pin.
+
+        .. note::
+            ``**kwargs`` ensures compatibility with additional arguments
+            introduced in future versions.
+
+        :param widget: The button that was pressed.
+        :param pin: The pin that was selected.
+        """
+        ...
+
+
+class MapView(Widget):
+    def __init__(
+        self,
+        id=None,
+        style=None,
+        location: toga.LatLng | None = None,
+        zoom: int | None = 2,
+        pins: list[MapPin] | None = None,
+        on_select: toga.widgets.mapview.OnSelectHandler | None = None,
+    ):
+        """Create a new MapView widget.
+
+        :param id: The ID for the widget.
+        :param style: A style object. If no style is provided, a default style will be
+            applied to the widget.
+        :param location: The initial latitude/longitude where the map should be
+            centered. If not provided, the initial location for the map is Perth,
+            Australia.
+        :param zoom: The initial zoom level for the map.
```

Comment:
Awwww... but Perth is a *nice* place :-)

The primary reason to avoid (0,0) is that if you specify a high zoom level but not a location, (0,0) is the middle of the ocean; and zoom level 0 isn't a "world" zoom level - it's "nation" - so you see a good chunk of Africa, but not a "whole world" map. 

The "Zoom 2 on Perth" behavior is essentially a variant on what Apple Maps does - by default, Apple Maps enters on the Apple campus in Cupertino, at a level so you can see the whole Bay Area. That's enough to let you know that the map is working, but also enough of a signal that you've probably forgotten to set an initial value.

Reply:
OK, makes sense.